protostar/cairo/short_string.py

```diff
@@ -1,8 +1,21 @@
 CairoShortString = int
 
+LAST_ASCII_CHAR_DEC_VAL = 127
+
 
 def short_string_to_str(value: CairoShortString) -> str:
     """
     Reverse of starkware.cairo.lang.compiler.test_utils.short_string_to_felt
     """
-    return value.to_bytes(length=32, byteorder="big").decode("ascii").lstrip("\x00")
+    return _to_bytes(value).decode("ascii").lstrip("\x00")
+
+
+def is_decodable_to_ascii(value: int) -> bool:
+    for byte in _to_bytes(value):
+        if byte > LAST_ASCII_CHAR_DEC_VAL:
+            return False
+    return True
+
+
+def _to_bytes(value: int) -> bytes:
+    return value.to_bytes(length=31, byteorder="big")
```

Comment:
Isn't there technically a possibility that we will encounter a value that occupies more than 31 bytes? Currently  P = 2^251 + 17 * 2^192 + 1 so in the worst case the felt passed there could be 252 bits = 31.5 bytes long. Of course it wouldn't happen if someone passes there a short string literal (since every ascii character is encoded using one whole byte), but if the program panics with some large felt252 like P - 1 we can get an error here. I think we should just set length to 32

Reply:
I guess there is no reason to convert large felt252 which is not a short string to string. Furthermore, it could contain bytes that are not decodable to ascii. If the felt is a short string, then it contains 31 bytes of meaningful information. If it's not then there is no reason to convert it to ascii string (and this conversion can fail)